### PR TITLE
torchmetrics 1.4.0.post0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-2732](https://anaconda.atlassian.net/browse/PKG-2732) 
- [Upstream repository](https://github.com/Lightning-AI/torchmetrics/tree/v1.4.0.post0)
- Requirements:
  - https://github.com/Lightning-AI/torchmetrics/blob/v1.4.0.post0/pyproject.toml
  - https://github.com/Lightning-AI/torchmetrics/blob/v1.4.0.post0/setup.py
  - https://github.com/Lightning-AI/torchmetrics/blob/v1.4.0.post0/requirements/base.txt

### Explanation of changes:

- Reset the build number
- Update runtime pinnings

### Notes:

- Updating because of the `pytorch 2.3.0` release

[PKG-2732]: https://anaconda.atlassian.net/browse/PKG-2732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,8 +23,8 @@ requirements:
   run:
     - python
     - numpy >1.20.0
-    - pytorch >=1.8.1
-    - lightning-utilities >=0.8.0
+    - pytorch >=1.10.0,<2.4.0
```
Comment (danpetry) [inline recipe/meta.yaml]: same with lightning, you don't need the upper bound

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Done

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,7 +23,7 @@ requirements:
   run:
     - python
     - numpy >1.20.0
-    - pytorch >=1.8.1
+    - pytorch >=1.10.0
     - lightning-utilities >=0.8.0
```
Comment (cbouss) [inline recipe/meta.yaml:27]:
```suggestion
    - packaging >17.1
    - lightning-utilities >=0.8.0
```
https://github.com/Lightning-AI/torchmetrics/blob/v1.4.0.post0/requirements/base.txt#L5

**Review by JamesRobertsonGames — APPROVED** [review]

File: recipe/meta.yaml [hunk 3]
```diff
@@ -24,7 +23,7 @@ requirements:
...
     - typing-extensions  # [py<39]
```
Comment (cbouss) [inline recipe/meta.yaml:28]: Can you add run_constrained with reqs from audio.txt, detection.txt, image.txt, multimodal.txt, text.txt and visual.txt?
https://github.com/Lightning-AI/torchmetrics/tree/v1.4.0.post0/requirements
https://inspector.pypi.io/project/torchmetrics/1.4.0.post0/packages/6d/e6/e51997d1818a4c1a1ad2b1c7ca5ff9dd95969596add58b2ed39479026964/torchmetrics-1.4.0.post0-py3-none-any.whl/torchmetrics-1.4.0.post0.dist-info/METADATA#line.28

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:27]: `packaging` isn't used outside their CI on github https://github.com/search?q=repo%3ALightning-AI%2Ftorchmetrics%20packaging&type=code

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:28]: Done

Comment (danpetry) [reply recipe/meta.yaml:28]: Again, upper bounds aren't relevant, please remove

Comment (skupr-anaconda) [reply recipe/meta.yaml:28]: Done. I left the upper bounds if there are no lower bounds

**Review by lorepirri — APPROVED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (danpetry) [reply recipe/meta.yaml:28]: just wondering what's the reasoning - in that case maybe there should just be no bounds at all?
[worth a glance over](https://iscinumpy.dev/post/bound-version-constraints/) - tl;dr is that upper bounds should be used sparingly and with consideration

**Pull request merged**